gdal.cmake

```diff
@@ -197,12 +197,15 @@ endif ()
 
 set(_CMAKE_C_FLAGS_backup ${CMAKE_C_FLAGS})
 set(_CMAKE_CXX_FLAGS_backup ${CMAKE_CXX_FLAGS})
+
 if (CMAKE_C_FLAGS)
-  string(REPLACE "-Werror" " " CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+  string(REGEX REPLACE "-Werror$" " " CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
+  string(REGEX REPLACE "-Werror " " " CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
   string(REPLACE "/WX" " " CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
 endif ()
 if (CMAKE_CXX_FLAGS)
-  string(REPLACE "-Werror" " " CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  string(REGEX REPLACE "-Werror$" " " CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+  string(REGEX REPLACE "-Werror " " " CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
```

Comment:
I would prefer quotes on the input, and an extra space for a single simple `REPLACE`:
~~~cmake
string(REPLACE "-Werror " " " CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ")
~~~
(`CMAKE_<LANG>_FLAGS` is a string, not a CMake list.)

Reply:
done